library/VersionInfoFactory.cpp

```diff
@@ -133,18 +134,30 @@ void VersionInfoFactory::ParseVersion (TiXmlElement* entry, VersionInfo* mem)
             if(!cstr_key)
                 throw Error::SymbolsXmlUnderspecifiedEntry(cstr_name);
             const char *cstr_value = pMemEntry->Attribute("value");
-            if(!cstr_value)
+            const char *cstr_mangled = pMemEntry->Attribute("mangled");
+            if(!cstr_value && !cstr_mangled)
             {
                 cerr << "Dummy symbol table entry: " << cstr_key << endl;
                 continue;
             }
             if ((is_vtable && no_vtables) || (!is_vtable && no_globals))
                 continue;
-#ifdef DFHACK64
-            uintptr_t addr = strtoull(cstr_value, 0, 0);
-#else
-            uintptr_t addr = strtol(cstr_value, 0, 0);
-#endif
+            uintptr_t addr;
+            if (cstr_value) {
+                if (sizeof(addr) == sizeof(unsigned long))
```

Comment:
This is bad on Windows because `sizeof(unsigned long) == 4` regardless of the architecture.

Reply:
else branch uses strtoull if sizeof unsigned long doesn't match sizeof uintptr_t. That means 32bit unix and windows uses the true branch calling strtoul which returns 32bit integer. In 64bit unix it also takes true branch and calls strtoul but this time the function returns 64bit integer. Then 64bit windows takes else branch and calls stroull which also returns 64 bit integer.

It would be possible to write c++ template function for all stdto* functions which would just check signens and sizeof integer. But this code uses only two alternatives so I kept the check inline.

There appears to be conflicts. I can rebase this later today.

Reply:
Oh, right, strtoul(l) will return something matching sizeof(addr). Makes sense